scipy/ndimage/_backend.py

```diff
@@ -0,0 +1,270 @@
+import scipy._lib.uarray as ua
+from scipy.ndimage import _api
+import numpy as np
+
+
+__all__ = ['register_backend', 'set_backend',
+           'set_global_backend', 'skip_backend']
+
+
+class ndimage_output:
+    """
+    Special case outputs to handle multiple types in __ua_convert__.
+    Eg: output='f', output='float32', output=np.float32
+        are all acceptable in addition to arrays.
+    """
+    pass
+
+
+class ndimage_index:
+    """
+    Special case index argument in measurement methods to
+    handle both int and array types in __ua_convert__.
+    """
+    pass
+
+
+class _ScipyImageBackend:
+    """The default backend for ndimage calculations
+
+    Notes
+    -----
+    We use the domain ``numpy.scipy`` rather than ``scipy`` because ``uarray``
+    treats the domain as a hierarchy. This means the user can install a single
+    backend for ``numpy`` and have it implement ``numpy.scipy.ndimage`` as
+    well.
+
+    .. versionadded:: 1.8.0
+    """
+    __ua_domain__ = "numpy.scipy.ndimage"
+
+    @staticmethod
+    def __ua_function__(method, args, kwargs):
+        fn = getattr(_api, method.__name__, None)
+
+        if fn is None:
+            return NotImplemented
+        return fn(*args, **kwargs)
+
+    @ua.wrap_single_convertor
+    def __ua_convert__(value, dispatch_type, coerce):
+        if value is None:
+            return None
+
+        if dispatch_type is np.ndarray:
+            if not coerce and not isinstance(value, np.ndarray):
+                return NotImplemented
+
+            return np.asarray(value)
+
+        if dispatch_type is np.dtype:
+            try:
+                return np.dtype(str(value))
+            except TypeError:
+                return np.dtype(value)
```

Comment:
Is there a reason this can't just be simplified to the following?:
```suggestion
            return np.dtype(value)
```

I see that this same try/except [is also in unumpy](https://github.com/Quansight-Labs/unumpy/blob/97dcc82e8f3343c2a66b5ee6d8081886652de1d0/unumpy/numpy_backend.py#L70-L74), but I am not sure why.  If you do change it here, please update the example in the `__ua_convert__` docs as well.

Reply:
Not sure why it was there in the first place. Seems fine to me refactoring this. Done.